SMLHelper/Handlers/EquipmentHandler.cs

```diff
@@ -0,0 +1,82 @@
+﻿namespace SMLHelper.V2.Handlers
+{
+    using SMLHelper.V2.Handlers;
+    using SMLHelper.V2.Interfaces;
+    using SMLHelper.V2.Patchers.EnumPatching;
+    using SMLHelper.V2.Utility;
+
+    /// <summary>
+    /// A handler class for everything related to creating new Equipments.
+    /// </summary>
+    public class EquipmentHandler : IEquipmentHandler
+    {
+        /// <summary>
+        /// Main entry point for all calls to this handler.
+        /// </summary>
+        public static IEquipmentHandler Main { get; } = new EquipmentHandler();
+
+        private EquipmentHandler()
+        {
+            // Hide constructor
+        }
+
+        /// <summary>
+        /// Adds a new <see cref="Equipment" /> into the game.
+        /// </summary>
+        /// <param name="equipmentName">The name of the Equipment. Should not contain special characters.</param>
+        /// <param name="displayName">The display name of the Equipment. Can be anything.</param>
+        /// <returns>
+        /// The new <see cref="Equipment" /> that is created.
+        /// </returns>
+        public EquipmentType RegisterNewEquipmentType(string equipmentName, string displayName)
+        {
+            EquipmentType equipment = EquipmentTypePatcher.AddEquipmentType(equipmentName);
+            LanguageHandler.SetLanguageLine(equipmentName, displayName);
+            return equipment;
+        }
+
+        /// <summary>
+        /// Safely looks for a modded group from another mod in the SMLHelper EquipmentCache.
+        /// </summary>
+        /// <param name="equipmentString">The string used to define the techgroup.</param>
+        /// <returns>
+        ///   <c>True</c> if the item was found; Otherwise <c>false</c>.
+        /// </returns>
+        public bool ModdedEquipmentExists(string equipmentString)
+        {
+            EnumTypeCache cache = EquipmentTypePatcher.cacheManager.RequestCacheForTypeName(equipmentString, false);
+            if (cache != null) // Item Found
+            {
+                return true;
+            }
+            else // Mod not present or not yet loaded
+            {
+                return false;
+            }
```

Comment:
These types of blocks can be simplified
```suggestion
            // if we don't have it cached, the mod is not present or not yet loaded
            return cache != null;
```
This is really minor though so I wouldn't really worry about it.

Reply:
I ensured consistency in the names, and I've also added this little optimization. I was a bit hasty, else I would have noticed it myself!